src/main/java/sparksoniq/spark/iterator/flowr/ReturnClauseSparkIterator.java

```diff
@@ -55,9 +57,36 @@ protected boolean initIsRDD() {
     public JavaRDD<Item> getRDD(DynamicContext context) {
 
         RuntimeIterator expression = this._children.get(0);
-        itemRDD = this._child.getRDD(context).flatMap(new ReturnFlatMapClosure(expression));
+        Dataset<Row> df = this._child.getDataFrame(context);
+        JavaRDD<Item> rdd = df.javaRDD().flatMap(new ReturnClauseDeserializeClosure());
+
+
+        // TODO: apply return expression
+        // itemRDD = rdd.flatMap(new ReturnFlatMapClosure(expression));
+        return rdd;
+    }
+    /*
+    public JavaRDD<Item> getDataFrame(DynamicContext context) {
```

Comment:
The return clause doesn't return any dataframes! It still returns an RDD of Items as before (but will consume the dataframe output by its child clause, if any) :-)

DataFrames only replace RDDs of tuples.